src/main/scala/mesosphere/marathon/core/group/impl/GroupManagerActor.scala

```diff
@@ -268,11 +268,13 @@ private[impl] class GroupManagerActor(
       to.transitiveApps.map {
         // assign values for service ports that the user has left "blank" (set to zero)
         case app: AppDefinition if app.hasDynamicServicePorts => assignPorts(app)
-        case app: AppDefinition =>
+        case app: AppDefinition if app.servicePorts.nonEmpty =>
           // Always set the ports to service ports, even if we do not have dynamic ports in our port mappings
+          // TODO: This is almost definitely wrong.
```

Comment:
Please you add your name/github-id to the `TODO(jasongilanfarr): ...` so we can attribute these later, unless you're planning on fixing this before we merge.